code/_helpers/_lists.dm

```diff
@@ -206,6 +206,20 @@ proc/listclearnulls(list/list)
 		result = first - second
 	return result
 
+/*
+Two lists may be different (A!=B) even if they have the same elements.
+This actually tests if they have the same entries and values.
+*/
+/proc/same_entries(var/list/first, var/list/second)
+	if(!islist(first) || !islist(second))
+		return 0
+	if(length(first) != length(second))
+		return 0
+	for(var/entry in first)
```

Comment:
Not sure what you mean. All lists are numerically indexed. You can access any list by doing listname[indexval]. Also this is a copypaste from Bay I believe.

The proc checks:
- If at least one list is not defined, they aren't the same
- If the lengths are different, they aren't the same
- If every key in first is present in second, and, if the values for said keys are the same, otherwise they're not the same
- Else the are they same

Reply:
This iterates over the contents of the list itself.  This proc checks that the lists have the same _entries_ not the same _order_.  For example `list("A", "B", "C")`  has the same entries as `list("B", "A", "C")`